src/engine/util/webxr/raycaster.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  Raycaster,
+  Vector3,
+  BufferGeometry,
+  LineBasicMaterial,
+  Line,
+  Matrix4,
+} from "three";
+
+/** ThreeJS Line, extended with an Update method */
+class SCLine extends Line {
+  constructor(
+    bufferGeo: THREE.BufferGeometry,
+    material: THREE.LineBasicMaterial
+  ) {
+    super(bufferGeo, material);
+  }
+  Update(): void {}
+}
+
+/**
+ * Sandcastle's own Raycaster; WebXR-friendly syntactic sugar over threeJS's Raycaster
+ * @extends Raycaster
+ */
+export class SCRaycaster extends Raycaster {
+  _originObject: THREE.Mesh | THREE.Group;
+  _target: THREE.Mesh | Array<THREE.Mesh> | THREE.Box3;
+  _direction: THREE.Vector3;
+  _isRecursive: boolean;
+  _near: number;
+  _far: number;
+  _visualizedRaycast: THREE.Line | undefined;
+  _isTargetArray: boolean;
+  _isTargetBox3: boolean;
+  _tempMatrix: THREE.Matrix4;
+  /**
+   * Construct a Sandcastle Raycaster.
+   * @param originObject - object to raycast from
+   * @param target - object(s) to raycast to. Can be a mesh, a mesh array or a Box3.
+   * @param [direction] - The normalized direction vector that gives direction to the ray.
+   * @param [isRecursive] -  If true, it also checks all descendants. Otherwise it only checks intersection with the object. Default is true.
+   * @param [near] - All results returned are further away than near. Near can't be negative. Default value is 0.1.
+   * @param [far] - All results returned are closer than far. Far can't be lower than near. Default value is 10.
+   */
+  constructor(
+    originObject: THREE.Mesh | THREE.Group,
+    target: THREE.Mesh | Array<THREE.Mesh> | THREE.Box3, // due to original polymorphism in THREE.Raycaster(), see below
```

Comment:
The inline comment might be better in the param description above

Reply:
actually no longer needed (it was written to explain the `any`, now the signature is self-documenting + JSDoc comments - thanks for the callout!